core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
@nmittler, I don't think you understand the intent of my quote. As it stands, if this PR went in as-is, I could make another PR that added back the `Preconditions`. Why wouldn't I do that, if _I_ like having the `Preconditions` more? It seems you don't have an argument as far as why that would be bad, but I do assert it _needs_ to be bad. For things where there are multiple reasonable options, we can't simply let it flip-flop between people's preferences. I felt that is dealt with by having some respect for the current state.

That isn't to say we couldn't change all the `checkNotNull`s. If we decided that not having `Preconditions` was better, then it would be fine to go and change the existing code. Similarly, if someone got bit by a `NullPointerException` that didn't have the message and took a lot of debugging to find, then adding the message could be reasonable.